src/etc/inc/interfaces.inc

```diff
@@ -1050,6 +1050,10 @@ function interface_bring_down($interface = "wan", $ifacecfg = false)
                         break;
                     }
                 }
+                if ( $ifcfg['ipaddrv6'] == 'slaac' || $ifcfg['ipaddrv6'] == 'dhcp6' ) {
+                  interface_dhcpv6_prepare($interface, $ifcfg, true);
+                  killbypid('/var/run/dhcp6c.pid', 'HUP');
+                }
                 return;
```

Comment:
what if we remove the return instead and let the code below handle this?

Reply:
second thought: maybe it would be better to kill ipv6 first (switch the blocks)

Reply:
In answer to the first comment, I'd need to run it up and see what happens; in answer to the second, explain, switch which blocks?

There's another REALLY weird issue I came across this morning. I'm on IRC

Reply:
IPv4 and IPv6 destroy blocks... I mean we set up IPv4 then IPv6, so bring down should kill IPv6 first, then IPv4?

In that case the return wouldn't be harmful because we already called the proper IPv6 destruction code. Unsure which is better, maybe both (switch blocks, no return) would be the most favourable.

I'm chasing down the origin of the "return" now to be sure.

Reply:
Ok, it's one of ours 31282787dfb7 ... need to discuss internally